hikari/utilities/iterators.py

```diff
@@ -941,6 +975,9 @@ async def _generator(self) -> typing.AsyncIterator[AnotherValueT]:
         async for input_item in self._iterator:
             result_iterator = self._flattener(input_item)
 
+            if isinstance(result_iterator, typing.Iterable):
+                result_iterator = iter(result_iterator)
+
             if isinstance(result_iterator, typing.Iterator):
```

Comment:
Doesn't iterator subclass Iterable meaning this would never be true as iterators would be caught by the iterable check?

Reply:
TIL.
```suggestion
            if isinstance(result_iterator, typing.Iterator):
```